calculator.js

```diff
@@ -0,0 +1,148 @@
+  base.appendChild(calculatorWrap);
+
+  // Add input box
+  var box = document.createElement('input');
+  box.className = 'cal-box';
+  box.id = 'cal-box';
+  calculatorWrap.appendChild(box);
+
+  // Add calculator buttons
+  for(var i = 0; i < buttons.length; i++) {
+    for(var j = 0; j < buttons[i].length; j++) {
+      var buttonContent = buttons[i][j];
+      var button = document.createElement('button');
+      button.className = 'cal-button';
+      button.id = buttonContent;
+      button.appendChild(document.createTextNode(buttonContent));
+      button.addEventListener('click', function(event) {
+        // If clicked button is a number
+        if(!isNaN(event.target.id)) {
+          // If last operation was math.eval then start new string
+          if(lastNum == 'new') {
+            box.value = '';
+          }
+          // Append new value
+          box.value = box.value + event.target.id;
+          lastNum = event.target.id;
+        // If clicked button is not a number
+        } else {
+          // If clicked button is '='
+          if(event.target.id == '=') {
+            // Evaluate current string
+            box.value = math.eval(box.value);
+            // Note that the last operation was an eval
+            lastNum = 'new';
+          // If clicked button is an operation
+          } else {
+            // If the last button clicked was a number, allow the operator to be appended
+            if(!isNaN(lastNum)) {
+              box.value = box.value + event.target.id;
+              lastNum = event.target.id;
+            }
+          }
+        }
+        // Remove keyboard focus on button
+        this.blur();
+      }, false);
+      calculatorWrap.appendChild(button);
+    }
+  }
+  // If Enter/Return is pressed, click '=' to evaluate
+  document.addEventListener('keypress', function(key){
+    console.log(String.fromCharCode(key.charCode));
+    if(key.keyCode == 13){
+      document.getElementById('=').click();
+    } else if (/[0-9]|[/*-+=.]/.test(String.fromCharCode(key.charCode))) {
+      document.getElementById(String.fromCharCode(key.charCode)).click();
+    }
+  }, false);
+}
+
+function themeButtons(base) {
+  var themeWrapper = document.createElement('div');
+  themeWrapper.id = 'theme-changer';
+  base.appendChild(themeWrapper);
+
+  var toggle = document.createElement('button');
+  toggle.id = 'toggle';
+  toggle.className = 'toggle';
+  toggle.appendChild(document.createTextNode('Toggle Theme'));
+  toggle.addEventListener('click', switchTheme, false);
+  themeWrapper.appendChild(toggle);
+}
+
+function switchTheme() {
+  var current = document.getElementsByTagName("link").item(2);
+  if(theme == 'light') {
+    theme = 'dark';
+    changeCSS('themes/dark.css', 2);
+  } else {
+    theme = 'light';
+    changeCSS('themes/light.css', 2);
+  }
+}
+
+function changeCSS(cssFile, cssLinkIndex) {
+  var oldlink = document.getElementsByTagName("link").item(cssLinkIndex);
```

Comment:
Accessing this by index is SUPER brittle. You could instead use standard alternate stylesheets [`rel="alternate stylesheet"`](https://developer.mozilla.org/en-US/docs/Web/CSS/Alternative_style_sheets) with unique `title` attributes.

You can find the alternate stylesheets in the document with querySelector (like `document.querySelector("link[title='light']")`) and toggle the `disabled` attribute to enable or disable them.